R/visualisation_recipe_internal.R

```diff
@@ -68,7 +72,35 @@
   # CI
   aes <- .find_aes_ci(aes, data)
 
-  list(aes = aes, data = data)
+  # axis and legend labels
+  if (!is.null(model_data) && !is.null(model_response)) {
+    # response - mapped to the y-axis
+    ylab <- .safe(attr(model_data[[model_response]], "label", exact = TRUE))
+    # fix default y-label, if necessary
+    y_prefix <- aes$y
+    if (y_prefix == "Predicted") {
+      y_prefix <- "Predicted value"
+    }
+    # set y-label based on labelled data, or variable name
+    if (is.null(ylab)) {
+      ylab <- paste(y_prefix, "of", model_response)
+    } else {
+      ylab <- paste(y_prefix, "of", ylab)
+    }
+    # main predictor - mapped to x-axis
+    xlab <- .safe(attr(model_data[[by[1]]], "label", exact = TRUE))
+    # first grouping variable (2nd in "by") - mapped to legend
+    if (length(by) > 1) {
+      colour <- .safe(attr(model_data[[by[[2]]]], "label", exact = TRUE))
+    } else {
+      colour <- NULL
+    }
+    axis_labels <- list(ylab = ylab, xlab = xlab, colour = colour)
+  } else {
+    axis_labels <- NULL
+  }
+
+  list(aes = aes, data = data, axis_labels = axis_labels)
```

Comment:
rather than a separate list, might be tidier to just add `xlab` and `ylab` and `colourlab` inside the first `aes` list

Reply:
Ok, will do